### ISSUE TITLE
 value dataset

### ISSUE THREAD

**Issue by mygithubforlst** [wgcban/ChangeFormer#53]
hello，could  you  please tell me why do the evaluation dataset in training and the dataset during testing use "test?"

**Comment by zaidbhat1234** [comment]
@mygithubforlst You can check here that training and test scripts use different loaders: https://github.com/wgcban/ChangeFormer/blob/9025e26417cf8f10f29a48f34a05758498216465/utils.py#L32
Train loader has been hard coded to use the 'val' split. I had the same question then I figured out they hard-code the dataloader to use val split in training.

**Comment by mygithubforlst** [comment]
Thank you very much for your reply. I have also noticed this part of the code, but if you re specify split_ Va in the. sh file, according to this part of the code, the split_val will be changed to the content specified in. sh instead of the original val

split_val = 'val'
if hasattr(args, 'split_val'):#hasattr() 
        split_val = args.split_val#in .sh file  split_val=test    this will useed as val datasets

so I don't quite understand
@zaidbhat1234

**Comment by wgcban** [comment]
https://github.com/wgcban/ChangeFormer/blob/9025e26417cf8f10f29a48f34a05758498216465/scripts/run_ChangeFormer_LEVIR.sh#L31-L32

split=train means train split is used for training and split_val=test means accuracy is computed on the test set.

**Comment by wgcban** [comment]
if you want to evaluate the model performance on validation set, specify the split_val = val
